// api/mockPhones.js
export default [
  {
    id: "1",
    categoryId: "1",
    name: "Apple iPhone 5c",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 823,
    image: "/uploads/iphone5c-selection-hero-2013.png",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/iphone5c-selection-hero-2013.png',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/iphone5c-selection-hero-2013.png',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/iphone5c-selection-hero-2013.png',
      name: 'front view'
    }]
  },
  {
    id: "2",
    categoryId: "1",
    name: "Apple iPhone 6",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 953,
    image: "/uploads/51u6y9Rm8QL._SY300_.jpg",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/51u6y9Rm8QL._SY300_.jpg',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/51u6y9Rm8QL._SY300_.jpg',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/51u6y9Rm8QL._SY300_.jpg',
      name: 'front view'
    }]
  },
  {
    id: "3",
    categoryId: "4",
    name: "Lenovo A6000",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 764,
    image: "/uploads/_35%20(1).JPG",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/_35%20(1).JPG',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/_35%20(1).JPG',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/_35%20(1).JPG',
      name: 'front view'
    }]
  },
  {
    id: "4",
    categoryId: "5",
    name: "Nokia Lumia 1520",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 674,
    image: "/uploads/Lumia1520-Front-Back-png.png",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/Lumia1520-Front-Back-png.png',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/Lumia1520-Front-Back-png.png',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/Lumia1520-Front-Back-png.png',
      name: 'front view'
    }]
  },
  {
    id: "5",
    categoryId: "3",
    name: "HTC One",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 674,
    image: "/uploads/htc-one-m7-802w-dual-sim-silver.jpg",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/htc-one-m7-802w-dual-sim-silver.jpg',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/htc-one-m7-802w-dual-sim-silver.jpg',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/htc-one-m7-802w-dual-sim-silver.jpg',
      name: 'front view'
    }]
  },
  {
    id: "6",
    categoryId: "2",
    name: "Samsung Galaxy",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam ultricies lorem odio, at laoreet tellus sodales in. Nullam maximus eros ut tortor ultricies rutrum. Aliquam euismod lacus non est egesta",
    price: 674,
    image: "/uploads/Agnes_Case_for_Samsung_Galaxy_S6_(1)__92643_thumb.jpg",
    cpu: "1.3GHz Apple A6",
    camera: "8mp (3264x2448)",
    size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
    weight: "132 grams (4.7 ounces) with battery",
    display: "4.0 326 pixel density",
    battery: "1480 mAh",
    memory: "16GB, 32GB and RAM 1 GB",
    images: [{
      id: "0",
      link: '/uploads/Agnes_Case_for_Samsung_Galaxy_S6_(1)__92643_thumb.jpg',
      name: 'front view'
    }, {
      id: "1",
      link: '/uploads/Agnes_Case_for_Samsung_Galaxy_S6_(1)__92643_thumb.jpg',
      name: 'front view'
    }, {
      id: "2",
      link: '/uploads/Agnes_Case_for_Samsung_Galaxy_S6_(1)__92643_thumb.jpg',
      name: 'front view'
    }]
  }
]
